### PR TITLE
osd/PeeringState: remove rollingforwarding from activate

### PR DESCRIPTION
**Pull request by ideepika**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by jdurgin — APPROVED** [review]
Let's see how this does in testing (esp. the ec overwrites tests given the history of the removed lines)

**Comment by ideepika** [comment]
testing subset ec; 4+2 pool: http://pulpito.front.sepia.ceph.com/ideepika-2021-03-10_12:38:09-rados:thrash-erasure-code-wip-tracker-48613-distro-basic-gibba/

**Comment by ideepika** [comment]
> testing subset ec; 4+2 pool: http://pulpito.front.sepia.ceph.com/ideepika-2021-03-10_12:38:09-rados:thrash-erasure-code-wip-tracker-48613-distro-basic-gibba/
> can run complete rados suite if it passes

hey @athanatos I see failure related to CopyFrom operation, do you have some idea if it could be somehow related to this change?

**Comment by athanatos** [comment]
It's not obvious to me.  What debugging have you done so far?

**Comment by neha-ojha** [comment]
> testing subset ec; 4+2 pool: http://pulpito.front.sepia.ceph.com/ideepika-2021-03-10_12:38:09-rados:thrash-erasure-code-wip-tracker-48613-distro-basic-gibba/
> can run complete rados suite if it passes

5953057 is the same as what I saw in my run, https://pulpito.ceph.com/nojha-2021-03-11_01:02:46-rados-wip-cc-weight-distro-basic-gibba/5954647/, looks like a new issue in master

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-14_09:40:23-rados-wip-kefu-testing-2021-03-14-1612-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-03-15_02:29:57-rados-wip-kefu-testing-2021-03-15-0034-distro-basic-smithi/

rerunning the failed test with master.

- https://pulpito.ceph.com/kchai-2021-03-15_09:33:05-rados-master-distro-basic-smithi/

**Comment by ideepika** [comment]
> * https://pulpito.ceph.com/kchai-2021-03-14_09:40:23-rados-wip-kefu-testing-2021-03-14-1612-distro-basic-smithi/
> * https://pulpito.ceph.com/kchai-2021-03-15_02:29:57-rados-wip-kefu-testing-2021-03-15-0034-distro-basic-smithi/
> * https://pulpito.ceph.com/kchai-2021-03-17_03:12:47-rados-wip-kefu-testing-2021-03-16-2006-distro-basic-smithi/
> 
> rerunning the failed test with master.
> 
> * https://pulpito.ceph.com/kchai-2021-03-15_09:33:05-rados-master-distro-basic-smithi/

reran, rerun of initial batch: https://pulpito.ceph.com/kchai-2021-03-15_02:29:57-rados-wip-kefu-testing-2021-03-15-0034-distro-basic-smithi/ with --filter-out="dashboard,cephadm" 
this pr: http://qa-proxy.ceph.com/teuthology/ideepika-2021-03-16_10:43:51-rados-wip-tracker-48613-distro-basic-smithi/5970615/teuthology.log

with patch:
```
+  if (pool.info.min_size >= info.stats.avail_no_missing.size() ) {
+    psdout(10) << " avail_no_missing size "  << info.stats.avail_no_missing.size() << dendl;
+    PGLog::LogEntryHandlerRef rollbacker{pl->get_log_handler(t)};
+    pg_log.roll_forward(rollbacker.get());
+  }
```
http://qa-proxy.ceph.com/teuthology/ideepika-2021-03-17_08:00:31-rados-wip-tracker-48613-distro-basic-smithi/5973818/teuthology.log

Kefu's failed+dead jobs rerun against master: 
- https://pulpito.ceph.com/kchai-2021-03-15_09:33:05-rados-master-distro-basic-smithi/

`qa/workunits/rados/test.sh` seems to be consistently failing other tests passed on rerun

**Comment by neha-ojha** [comment]
> > * https://pulpito.ceph.com/kchai-2021-03-14_09:40:23-rados-wip-kefu-testing-2021-03-14-1612-distro-basic-smithi/
> > * https://pulpito.ceph.com/kchai-2021-03-15_02:29:57-rados-wip-kefu-testing-2021-03-15-0034-distro-basic-smithi/
> > * https://pulpito.ceph.com/kchai-2021-03-17_03:12:47-rados-wip-kefu-testing-2021-03-16-2006-distro-basic-smithi/
> > 
> > rerunning the failed test with master.
> > 
> > * https://pulpito.ceph.com/kchai-2021-03-15_09:33:05-rados-master-distro-basic-smithi/
> 
> reran, rerun of initial batch: https://pulpito.ceph.com/kchai-2021-03-15_02:29:57-rados-wip-kefu-testing-2021-03-15-0034-distro-basic-smithi/ with --filter-out="dashboard,cephadm"
> this pr: http://qa-proxy.ceph.com/teuthology/ideepika-2021-03-16_10:43:51-rados-wip-tracker-48613-distro-basic-smithi/5970615/teuthology.log
> 
> with patch:
> 
> ```
> +  if (pool.info.min_size >= info.stats.avail_no_missing.size() ) {
> +    psdout(10) << " avail_no_missing size "  << info.stats.avail_no_missing.size() << dendl;
> +    PGLog::LogEntryHandlerRef rollbacker{pl->get_log_handler(t)};
> +    pg_log.roll_forward(rollbacker.get());
> +  }
> ```
> 
> http://qa-proxy.ceph.com/teuthology/ideepika-2021-03-17_08:00:31-rados-wip-tracker-48613-distro-basic-smithi/5973818/teuthology.log
> 
> Kefu's failed+dead jobs rerun against master:
> 
> * https://pulpito.ceph.com/kchai-2021-03-15_09:33:05-rados-master-distro-basic-smithi/
> 
> `qa/workunits/rados/test.sh` seems to be consistently failing other tests passed on rerun

which jobs exactly?

**Comment by ideepika** [comment]
> which jobs exactly?
batch: http://pulpito.front.sepia.ceph.com/ideepika-2021-03-17_08:00:31-rados-wip-tracker-48613-distro-basic-smithi/
http://qa-proxy.ceph.com/teuthology/ideepika-2021-03-17_08:00:31-rados-wip-tracker-48613-distro-basic-smithi/5973816/teuthology.log

**Comment by ideepika** [comment]
run: https://pulpito.ceph.com/ideepika-2021-03-31_09:08:48-rados-wip-tracker-48613-distro-basic-gibba/
rerun: https://pulpito.ceph.com/ideepika-2021-03-31_19:22:50-rados-wip-tracker-48613-distro-basic-smithi/

File: src/osd/PeeringState.h [hunk 1]
```diff
@@ -2314,8 +2314,8 @@ class PeeringState : public MissingLoc::MappingInfo {
    * applicable stretch cluster constraints.
    */
   bool acting_set_writeable() {
-    return (acting.size() >= pool.info.min_size) &&
-      (pool.info.stretch_set_can_peer(acting, *get_osdmap(), NULL));
+    return (acting.size() >= pool.info.min_size)
+      && (pool.info.stretch_set_can_peer(acting, *get_osdmap(), NULL));
```
Comment (athanatos) [inline src/osd/PeeringState.h:2318]: Why the whitespace change?

**Review by athanatos — APPROVED** [review]
LGTM otherwise

**Comment by athanatos** [comment]
Hmm, I don't think this could fix https://tracker.ceph.com/issues/48417.  Going active (not peered) with a last_update of e means that we may not ever rollback any e' < e.  This seems like a reasonable simplification, but I don't think it can actually result in an unfound object.

**Comment by neha-ojha** [comment]
> run: https://pulpito.ceph.com/ideepika-2021-03-31_09:08:48-rados-wip-tracker-48613-distro-basic-gibba/
> rerun: https://pulpito.ceph.com/ideepika-2021-03-31_19:22:50-rados-wip-tracker-48613-distro-basic-smithi/

@ideepika https://pulpito.ceph.com/ideepika-2021-03-31_19:22:50-rados-wip-tracker-48613-distro-basic-smithi/6013628/ is not related to this PR and is tracked in https://tracker.ceph.com/issues/50106 now.

**Comment by neha-ojha** [comment]
> Hmm, I don't think this could fix https://tracker.ceph.com/issues/48417. Going active (not peered) with a last_update of e means that we may not ever rollback any e' < e. This seems like a reasonable simplification, but I don't think it can actually result in an unfound object.

Given that we saw an unnecessary rollforward caused due to this in our tests https://tracker.ceph.com/issues/48613#note-14, I am fine with this patch.

**Comment by athanatos** [comment]
@neha-ojha See my last comment on https://tracker.ceph.com/issues/48613, there seems to be something else going on.

**Pull request closed without merging**